### ISSUE TITLE
[Story] Select a step in the process for each associated financial institution

### ISSUE THREAD

**Issue by angelcardoz** [cfpb/sbl-project#127]
### Epic

Epic issue includes overview, complete list of user stories, technical and content requirements, user testing task ideation, and current design mock-ups.

[Epic] Small business lending filing home https://github.com/cfpb/sbl-project/issues/14

### User story

[Story] As a filer, I want to select a step in the process for each associated financial institution, so that I can continue working on a task that is in progress or review a task that is completed.

### Acceptance criteria

Given a system user is logged in and has started the filing process for an associated financial institution, when they visit the small business lending filing home page, then the system displays the tasks that are in progress and/or have been completed for the respective financial institution along with an actionable element, then when the user clicks on the element they are redirected to that task. 

### Technical tasks

Developers should create technical tasks and add links (below) prior to sprint planning.

#### Front end
- [ ] Task (link)
- [ ] Task (link)

#### Back end
- [ ] Task (link)
- [ ] Task (link)

**Comment by natalia-fitzgerald** [comment]
@angelcardoz @dan-padgett 
I'd like to dig in to the premise of this user story a bit more. I originally interpreted this user story simply indicating the broad filing status to a user - not started (no files have been uploaded), started (a file has been uploaded), or completed (filing signed and submitted). In the initial mock-ups this is indicated in the card heading, the card body text, and the button language. 

The acceptance criteria, as written, takes this a step further and indicates that the financial institution card must accommodate an update status for each of the 5 steps in the filing process. This adds complexity, both visually and technically. So, before we decide to move in this direction I want to be sure that we've confirmed that a more simple approach does not meet the needs of the user, particularly in thinking about MVP. 

In the current design there is a "History of progress this filing period" on the homepage that lists the actions that have been taken for a filing. It was determined recently that this is not needed for MVP. Additionally some determined that this level of detail would be more helpful at the upload stage, if we decided to later add it. 

The "History of progress this filing period" would provide a user with a detailed accounting of all of the actions that have happened within a filing period (and the names of the users within the FI that made those changes). In HMDA, and as originally designed for SBL, this means file uploads. But for SBL, it has recently been mentioned that actions could include non-upload actions, in other words any type of change that is made by the user within a step/task. The "History of progress" is not immediately visible and must be clicked on by the user to see it. But, if this contextual information is only needed for certain users (users that start and come back, users that have multiple authorized/associated filers within their FI - are these the users that would need this?). 

So, I'd like to dig a little deeper as to whether the user needs to see the 5 process steps and the status of each step on the filing home screen within the financial institution card? I wouldn't want to overwhelm a user or make the process seem more complicated or open ended than it actually is. Can we dig into this a bit more? Can we consider alternative places where we can give the user the opportunity to jump around while still providing more structure and guidance from the home card? 

Also, this story looks similar but perhaps not identical to: https://github.com/cfpb/sbl-project/issues/110

**Comment by natalia-fitzgerald** [comment]
> @natalia-fitzgerald Here's a recap of our discussion from this morning and then a proposed change to the acceptance criteria.
> 
> # Discussion Recap
> * HMDA more or less rolled up the FI details and lending data into a single artifact, which is qualitatively different from how SBL handles that information.
> * It's important that a filer can go in and review FI details independently of the file upload process (excepting considerations with LEI). But it's good to follow HMDA's precedent for the file upload process.
> * One way to accommodate this is to split the task of reviewing FI details from the rest of the filing tasks. This would result in two main actions a user can take once they chosen the FI: review FI details, and start/continue uploading their data. One thing we didn't touch on was whether adding a POC is a part of the data upload process, reviewing FI details, or should be it's own action.
> 
> Did I miss anything about where the discussion ended?
> 
> # Acceptance Criteria Changes
> If we want to move in that direction, I think the acceptance criteria should be changed as follows: Given a system user is logged in ~and has started the filing process for an associated financial institution~, when they visit the small business lending filing home page, then the system ~displays the tasks that are in progress and/or have been completed for the respective financial institution along with an actionable element~ allows them to review financial institution details or start/continue the file upload process, then when the user clicks on the element they are redirected to that task.
> 
> With this revision, both the POC and sign and submit steps fall under the file upload task flow. Once a user goes to the file upload process, they would see a progress indicator similar to what HMDA shows.

@dan-padgett - I think the user story will need some adjusting too. How's this? As a filer, I want to select the next step in the process for my associated financial institution, so that when I leave the platform and return I can continue my work where I left off.

**Comment by dan-padgett** [comment]
@natalia-fitzgerald Yes, I think changing the language from "select a step" to "select the next step" fits with how we've been evolved our thinking on this task. I'm not sure we need "when I leave the platform" in the story since this would also cover the case of a user starting the filing process (in my mind, "next step" include "first step").

**Comment by natalia-fitzgerald** [comment]
@dan-padgett 

Yep, makes sense. 

**User story:**  
As a filer, I want to select the next step in the filing process for my associated financial institution, so that I can progress toward the completion of my filing.

**Comment by natalia-fitzgerald** [comment]
@angelcardoz 
@dan-padgett and I worked on updating this. Feel free to adjust as needed. We have been trying to determine whether there is a secondary action for each primary action and whether there is any need for any more than 2 possible actions for each entry point. In HMDA, for example, in the Review edits step there is: "Review edits" (button), "Download edit report" (link) and "Upload a new file" (link).  

Generally we are trying to move toward guiding a user into the process (from the home screen) - through a more defined path - with the idea that if they would like to jump around and not follow the linear flow they have that option once they are in the filing app. Within the app they would be able to revisit steps they have "completed" but probably not be able to move ahead past uncompleted steps.   

**User story:**
As a filer, I want to select the next step in the filing process for my associated financial institution, so that I can progress toward the completion of my filing.

**Short name:**
Select the next step in the filing process for my associated financial institution

**Acceptance criteria**
Given a system user is logged in and has started the filing process for an associated financial institution, when they visit the small business lending filing home page, then the system displays the next step and an actionable element, then when the user clicks on the element they are redirected to that task.
